packages/teams-js/src/internal/videoEffectsUtils.ts

```diff
@@ -373,7 +426,10 @@ class TransformerWithMetadata {
    */
   private extractVideoFrameAndMetadata = async (
     texture: VideoFrame,
-  ): Promise<{ videoFrame: VideoFrame; metadata: { audioInferenceResult?: Uint8Array } }> => {
+  ): Promise<{
+    videoFrame: VideoFrame;
+    metadata: { audioInferenceResult?: Uint8Array; attributes?: ReadonlyMap<string, Uint8Array> };
```

Comment:
>? [](http://example.com/codeflow?start=61&length=1)

What is the expectation on the difference between when someone passes in `undefined` for this value (or just doesn't pass it at all) vs when they pass in a defined but empty map object?

Reply:
`attributes` will be `undefined` for hosts that do not (yet) support using the required capabilities specified by the app. Otherwise, it will be empty for apps that don't specify any required capabilities.